### ISSUE TITLE
How to implement a train on only 1 class?

### ISSUE THREAD

**Issue by shlykov2** [ShuangXieIrene/ssds.pytorch#36]
for mobilenetV2 ssd-lite on COCO dataset. want to train to detect only pedestrians. I am sure that changing "num_classes" arg in the yml file isn't enough. My guess is: change the COCO ann file to "person_keypoints_train/val2017.json" and num_classes to 2. Will it be enough? Thank you.

**Comment by holyhao** [comment]
@lupotto Hi, thanks for your reply. Recently，I am training a  two classes task (object or not). I want to add some images that do not contain any object as the background . so, how can I put these images in training, and the images that do not contain any object can calculate the loss well?  Looking forward your answer.

**Comment by holyhao** [comment]
@lupotto  Thanks for your reply. I am new about detection task and please correct me if I am wrong. You change the id of the objects that were not pedestrian as a background(class-0), so the bounding box of them do not used in training .Because the negatives are generated by the anchor proposals of the image with the pedestrian object. Sometimes, there is some condition that some object is very similar to pedestrian in a image which has no pedestrian object . We want to add this kind of image with boundingbox of the misleading object as negatives. But, as I say above,  negatives are generated by the anchor proposals in training.  So, how can we add those misleading object as negatives ?

**Comment by 1453042287** [comment]
@lupotto maybe an extra class would be help~

**Comment by lupotto** [comment]
@holyhao Sorry of the delay of the answer.

> We want to add this kind of image with boundingbox of the misleading object as negatives. But, as I say above, negatives are generated by the anchor proposals in training. So, how can we add those misleading object as negatives ?

You cannot add this misleading objects as positive or negative. In training, you just enter images in the network that are already labelled. Saying briefly, based on the difference between the detection and the ground truth you calculate the loss. Afterwards you optimize until convergence. So (I don't know if I am answering your question) you just have to do that:

> You change the id of the objects that were not pedestrian as a background(class-0), so the bounding box of them do not used in training .

If by negatives do you mean False-Positives, i.e. is detecting something that it shouldn't be detected, what you can do is set one threshold of confidence in postprocessing and simply avoid those that do not have a confidence score above some %. In your configuration file do you have this part:

```
POST_PROCESS:
  SCORE_THRESHOLD: 0.01
  IOU_THRESHOLD: 0.6
  MAX_DETECTIONS: 100
```
You may want to increase this score_threshold to 0.2/0.3/0.4 in order to have less false-positives (more confidence). Depending on your application it would be helpful, but it will affect your overall mAP because you will have less recall.

Hope to answer your question

**Comment by holyhao** [comment]
@lupotto  Yeah, it is really helpful and thanks for your kind reply.
